src/server/JasmineGraphServer.cpp

```diff
@@ -3156,294 +2216,41 @@ void JasmineGraphServer::initiateMerge(std::string graphID, std::string training
-
-    if (sockfd < 0) {
-        server_logger.log("Cannot create socket", "error");
-        return 0;
-    }
-
-    if (host.find('@') != std::string::npos) {
-        host = Utils::split(host, '@')[1];
-    }
-
-    server = gethostbyname(host.c_str());
-    if (server == NULL) {
-        server_logger.log("ERROR, can not find the host", "error");
-        return 0;
+        for (it = partitions.begin(); it < partitions.end(); it++) {
+            partition_count++;
+        }
     }
 
-    bzero((char *)&serv_addr, sizeof(serv_addr));
-    serv_addr.sin_family = AF_INET;
-    bcopy((char *)server->h_addr, (char *)&serv_addr.sin_addr.s_addr, server->h_length);
-    serv_addr.sin_port = htons(port);
-    if (Utils::connect_wrapper(sockfd, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
-        server_logger.log("ERROR connecting", "error");
-        // TODO::exit
-        return 0;
-    }
-
-    bzero(data, FED_DATA_LENGTH);
-    write(sockfd, JasmineGraphInstanceProtocol::HANDSHAKE.c_str(), JasmineGraphInstanceProtocol::HANDSHAKE.size());
-    server_logger.log("Sent fed : " + JasmineGraphInstanceProtocol::HANDSHAKE, "info");
-    bzero(data, FED_DATA_LENGTH);
-    read(sockfd, data, FED_DATA_LENGTH);
-    string response = (data);
-
-    response = Utils::trim_copy(response, " \f\n\r\t\v");
-
-    if (response.compare(JasmineGraphInstanceProtocol::HANDSHAKE_OK) == 0) {
-        server_logger.log("Received fed : " + JasmineGraphInstanceProtocol::HANDSHAKE_OK, "info");
-        string server_host = masterIP;
-        write(sockfd, server_host.c_str(), server_host.size());
-        server_logger.log("Sent fed : " + server_host, "info");
-        bzero(data, FED_DATA_LENGTH + 1);
-        read(sockfd, data, FED_DATA_LENGTH);
-        response = (data);
-        response = Utils::trim_copy(response, " \f\n\r\t\v");
-
-        if (response.compare(JasmineGraphInstanceProtocol::HOST_OK) == 0) {
-            server_logger.log("Received : " + JasmineGraphInstanceProtocol::HOST_OK, "info");
-        }
+    std::thread *workerThreads = new std::thread[partition_count + 1];
 
-        write(sockfd, JasmineGraphInstanceProtocol::INITIATE_SERVER.c_str(),
-              JasmineGraphInstanceProtocol::INITIATE_SERVER.size());
-        server_logger.log("Sent fed : " + JasmineGraphInstanceProtocol::INITIATE_SERVER, "info");
-        bzero(data, FED_DATA_LENGTH + 1);
-        read(sockfd, data, FED_DATA_LENGTH);
-        response = (data);
-        response = Utils::trim_copy(response, " \f\n\r\t\v");
-        if (response.compare(JasmineGraphInstanceProtocol::OK) == 0) {
-            server_logger.log("Received fed : " + JasmineGraphInstanceProtocol::OK, "info");
-            write(sockfd, (trainingArgs).c_str(), (trainingArgs).size());
-            server_logger.log("Sent fed : training args " + trainingArgs, "info");
-            bzero(data, FED_DATA_LENGTH);
-            return 0;
+    trainingArgs = trainingArgs;
```

Comment:
Let delete this unnecessary line `trainingArgs = trainingArgs;`

Reply:
Removed this and all such occurrences in next PR.

Reply:
Ok